lib/rspec/abq.rb

```diff
@@ -106,14 +106,6 @@ def self.setup_after_specs_loaded!
       # before abq can start workers, it asks for a manifest
       if !!ENV[ABQ_GENERATE_MANIFEST] # the abq worker will set this env var if it needs a manifest
         RSpec::Abq::Manifest.write_manifest(RSpec.world.ordered_example_groups, RSpec.configuration.seed, RSpec.configuration.ordering_registry)
-        # ... Maybe it's fine to just exit(0)
-        if Gem::Version.new(RSpec::Core::Version::STRING) >= Gem::Version.new("3.10.0")
-          RSpec.world.wants_to_quit = true # ask rspec to exit
```

Comment:
yes. Gimme a few moments and I'll write up some notes on why this is okay (besides the fact that the integration test suite is in a good place right now)

Reply:
`setup_after_specs_loaded!`'s caller has moved around a bit since its inception. 

Now it's in RSpec::Core::Runner#run_specs (#14)
it was in RSpec::Configuration#load_spec_files (#11)
Originally (from your implementation maybe?) it was in RSpec::Core::Runner#setup

In its current location, as soon as [`run_specs` returns 0](https://github.com/rwx-research/rspec-abq/blob/main/lib/rspec/abq/extensions.rb#L113-L115), the whole process will exit with status 0. The two callers are `[Runner#invoke](https://github.com/rspec/rspec-core/blob/522b7727d02d9648c090b56fa68bbdc18a21c04d/lib/rspec/core/runner.rb#L45-L47)` and `[Runner.run](https://github.com/rspec/rspec-core/blob/522b7727d02d9648c090b56fa68bbdc18a21c04d/lib/rspec/core/runner.rb#L64-L73)`.

[`#invoke` is the entry point for the binary](https://github.com/rspec/rspec-core/blob/main/exe/rspec#L4). I believe `#run` is only used in tests.



In order to avoid needing to require `rspec-abq` explicitly, we moved the extension

Reply:
Understood. Thank you